### ISSUE TITLE
Ensure training pipeline is deterministic

### ISSUE THREAD

**Issue by jackapbutler** [OpenBioML/chemnlp#253]
In order to decrease sources of error we want to make sure the training pipeline creates a deterministic ordering of the dataset batches and other elements regarding initialisation. This will help to track down the sources of loss spikes i.e. malformed samples.

See [this experiment](https://www.notion.so/stable-society/1B-300M-DS-configs-9879c506beb44800a6abbaf40beed749) for more information. We might want to check out [this](https://huggingface.co/docs/transformers/internal/trainer_utils#transformers.enable_full_determinism).

**Comment by jackapbutler** [comment]
In the current non-nightly releases of PyTorch the setting above causes training to hang and is a known issue https://github.com/huggingface/transformers/issues/22363, going to try `torch.backends.cudnn.deterministic = True`.

**Comment by jackapbutler** [comment]
Using `torch.backends.cudnn.deterministic = True` and `model.eval()` to counteract dropout the loss for each of the first N batches became more similar but not completely consistent.
